pkg/distributor/forwarding/forwarding.go

```diff
@@ -253,7 +262,25 @@ type tsByTargets map[string]tsWithSampleCount
 
 // copyToTarget copies the given time series into the given target and does the necessary accounting.
 // The time series is deep-copied, so the passed in time series can be returned to the pool without affecting the copy.
-func (t tsByTargets) copyToTarget(target string, ts mimirpb.PreallocTimeseries, pool *pools) {
+func (t tsByTargets) copyToTarget(target string, dontForwardBeforeTs int64, ts mimirpb.PreallocTimeseries, pool *pools) (err error) {
+	if dontForwardBeforeTs > 0 && samplesNeedFiltering(ts.TimeSeries.Samples, dontForwardBeforeTs) {
+		samplesUnfiltered := ts.TimeSeries.Samples
+		defer func() {
+			ts.TimeSeries.Samples = samplesUnfiltered
+		}()
+
+		samplesFiltered := filterSamplesBefore(samplesUnfiltered, dontForwardBeforeTs)
```

Comment:
`filterSamplesBefore` name is not clear to me ... does it keep or remove samples that are before given timestamp?

Reply:
I renamed it to `dropSamplesBefore` 75a569399